packages/amplify-graphql-transformer-migrator/src/schema-inspector.ts

```diff
@@ -0,0 +1,55 @@
+import { stateManager } from 'amplify-cli-core';
+import { DocumentNode } from 'graphql/language';
+import { visit } from 'graphql';
+import { collectDirectivesByTypeNames } from '@aws-amplify/graphql-transformer-core';
+import { listContainsOnlySetString } from './utils';
+import * as fs from 'fs-extra';
+
+
+export function graphQLUsingSQL(apiName: string): boolean {
+  const teamProviderInfo = stateManager.getTeamProviderInfo();
+  const env = stateManager.getLocalEnvInfo().envName;
+  if (teamProviderInfo && teamProviderInfo[env]) {
+    const apiCategory = teamProviderInfo[env]?.categories?.api;
+    if (apiCategory && apiCategory[apiName] && apiCategory[apiName]?.rdsClusterIdentifier) {
+      return true;
+    }
+  }
+  return false;
+}
+
+
+export function detectCustomResolvers(schema: DocumentNode): boolean {
+  let customResolversUsed = false;
+  visit(schema, {
+    ObjectTypeDefinition: {
+      enter(node) {
+        if (node.name.value === 'Mutation' || node.name.value === 'Query' || node.name.value === 'Subscription') {
+          customResolversUsed = true;
+        }
+      }
+    }
+  });
+  return customResolversUsed;
+}
+
+
+export function detectOverriddenResolvers(apiName: string): boolean {
+  const files = fs.readdirSync(`amplify/backend/api/${apiName}/resolvers/`);
```

Comment:
I believe `readdirSync()` will return an empty array if the directory is empty, so you don't need the `files &&` part of the check below.